hibernate-core/src/test/java/org/hibernate/test/hql/UpdateJoinedSubclassCorrelationTest.java

```diff
@@ -0,0 +1,91 @@
+package org.hibernate.test.hql;
+
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.hibernate.testing.TestForIssue;
+import org.junit.Assert;
+import org.junit.Test;
+
+import javax.persistence.*;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+
+import java.util.List;
+
+import static javax.persistence.InheritanceType.JOINED;
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+
+@TestForIssue(jiraKey = "")
+public class UpdateJoinedSubclassCorrelationTest extends BaseEntityManagerFunctionalTestCase {
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { Master.class, SubMaster.class, Detail.class };
+	}
+
+	@Test
+	public void testJoinedSubclassUpdateWithCorrelation() {
+		// prepare
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			Master m1 = new SubMaster( 1, null );
+			entityManager.persist( m1 );
+			Detail d11 = new Detail( 10, m1 );
+			entityManager.persist( d11 );
+		} );
+
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			// DO NOT CHANGE this query: it used to trigger a very specific bug caused
+			// by the root table alias being added to the generated subquery instead of the table name
+			String d = "update SubMaster m set name = (select 'test' from Detail d where d.master = m)";
+			Query del = entityManager.createQuery( d );
+			del.executeUpdate();
```

Comment:
Looks like you copied a test doing deletes `d`, `del`. Maybe you could change that?